src/rust/engine/sharded_lmdb/src/lib.rs

```diff
@@ -378,6 +381,120 @@ impl ShardedLmdb {
       .await
   }
 
+  ///
+  /// Stores the given Read instance under its computed digest. This method performs two passes
+  /// over the source to 1) hash it, 2) store it. If !data_is_immutable, the second pass will
+  /// re-hash the data to confirm that it hasn't changed.
+  ///
+  /// If the Read instance gets longer between Reads, we will not detect that here, but any
+  /// captured data will still be valid.
+  ///
+  pub async fn store<F, R>(
+    &self,
+    initial_lease: bool,
+    data_is_immutable: bool,
```

Comment:
This is the only scary part of the review. The eyeball glazes over the bool literals at the final call sites where this is plumbed. I like a bool enum to force spelling out what choice is being made more clearly, but maybe you consider that overkill here:
https://github.com/pantsbuild/pants/blob/7da328c125ee8c17ba343c6a7b0aaa6066dba17b/src/rust/engine/client/src/options/args.rs#L18-L21

Reply:
That said, if you're modifying this code and not reading closely in the 1st place, I'm sympathetic to the idea you'll get what you deserve.